src/Schema.php

```diff
@@ -67,7 +67,7 @@ public function getRelations(string $role): array
      */
     public function defines(string $role): bool
     {
-        return array_key_exists($role, $this->schema) || array_key_exists($role, $this->aliases);
+        return isset($this->schema[$role]) || isset($this->aliases[$role]);
```

Comment:
if `$this->schema[$role]` and `$this->aliases[$role]` are set to null then the role is not actually defined....so replacing with faster `isset` language-construct. (pinging @wolfy-j )

Reply:
I think this is going to be one of the hottest parts of the code. Perfect!